contrib/test_jenkins.sh

```diff
@@ -1,277 +1,484 @@
+		../contrib/buildrpm.sh -s -b
+	fi
+
+	$MAKE distclean
+}
+
+#
+# Build with Intel compiler
+#
+build_icc() {
+	echo 1..1 > build_icc.tap
+	if module_load intel/ics
+	then
+		echo "==== Build with Intel compiler ===="
+		../contrib/configure-devel --prefix=$ucx_inst CC=icc CXX=icpc
+		$MAKE clean
+		$MAKE
+		$MAKE distclean
+		module unload intel/ics
+		echo "ok 1 - build successful " >> build_icc.tap
+	else
+		echo "==== Not building with Intel compiler ===="
+		echo "ok 1 - # SKIP because Coverity not installed" >> build_icc.tap
+	fi
+}
+
+#
+# Compile and run UCP hello world example
+#
+run_ucp_hello() {
+	gcc -o ./ucp_hello_world ${ucx_inst}/share/ucx/examples/ucp_hello_world.c \
+		-lucp -lucs -I${ucx_inst}/include -L${ucx_inst}/lib
 
-echo "Build docs"
-make $make_opt docs
-
-echo "Running ucx_info"
-./src/tools/info/ucx_info -v -f
-./src/tools/info/ucx_info -c
-
-echo "Build without IB verbs"
-../contrib/configure-release --without-verbs
-make $make_opt
-
-if [ -n "$JENKINS_RUN_TESTS" ]; then
-    # Set CPU affinity to 2 cores, for performance tests
-    if [ -n "$EXECUTOR_NUMBER" ]; then
-        AFFINITY="taskset -c $(( 2 * EXECUTOR_NUMBER ))","$(( 2 * EXECUTOR_NUMBER + 1))"
-        TIMEOUT="timeout 160m"
-        TIMEOUT_VALGRIND="timeout 200m"
-    else
-        AFFINITY=""
-        TIMEOUT=""
-        TIMEOUT_VALGRIND=""
-    fi
-
-    # Load newer doxygen
-    module load tools/doxygen-1.8.11
-
-    echo "Build gtest"
-    module load hpcx-gcc
-
-    make $make_opt clean
-
-    # todo: check in -devel mode as well
-    ../contrib/configure-release --with-mpi --prefix=$ucx_inst
-    make $make_opt install
-
-    # Prevent out tests from using installed UCX libraries
-    export LD_LIBRARY_PATH=${ucx_inst}/lib:$LD_LIBRARY_PATH
-
-    # check whether installation is valid (it compiles examples at least)
-    make installcheck
-
-    ucx_inst_ptest=$ucx_inst/share/ucx/perftest
-
-    # hack for perftest, no way to override params used in batch
-    # todo: fix in perftest
-    sed -s 's,-n [0-9]*,-n 1000,g' $ucx_inst_ptest/msg_pow2 | sort -R > $ucx_inst_ptest/msg_pow2_short
-    cat $ucx_inst_ptest/test_types | sort -R > $ucx_inst_ptest/test_types_short
+	# hello-world example
+	tcp_port=$((10000 + EXECUTOR_NUMBER))
+	for test_mode in -w -f -b
+	do
+		echo "==== Running UCP hello world with mode ${test_mode} ===="
+		./ucp_hello_world ${test_mode} -p ${tcp_port} &
+		hw_server_pid=$!
+
+		sleep 5
+
+		# need to be ran in background to reflect application PID in $!
+		./ucp_hello_world ${test_mode} -n $(hostname) -p ${tcp_port} &
+		hw_client_pid=$!
+
+		# make sure server process is not running
+		wait ${hw_server_pid} ${hw_client_pid}
+	done
+	rm -f ./ucp_hello_world
+}
+
+#
+# Compile and run UCT hello world example
+#
+run_uct_hello() {
+	gcc -o ./uct_hello_world ${ucx_inst}/share/ucx/examples/uct_hello_world.c \
+		-luct -lucs -I${ucx_inst}/include -L${ucx_inst}/lib
+
+	tcp_port=$((10000 + ${EXECUTOR_NUMBER}))
```

Comment:
Code duplication with ucp_hello_world. It might be better to calculate the port once and use it for all required client/server applications in the test.